pkg/provider/aws/resources/api_gateway.go

```diff
@@ -85,6 +87,278 @@ type (
+	ApiName    string
+	HttpMethod string
+}
+
+// Create takes in an all necessary parameters to generate the RestApi name and ensure that the RestApi is correlated to the constructs which required its creation.
+func (integration *ApiIntegration) Create(dag *core.ResourceGraph, params ApiIntegrationCreateParams) error {
+
+	name := apiResourceSanitizer.Apply(fmt.Sprintf("%s-%s-%s", params.AppName, params.Name, params.HttpMethod))
+	integration.Name = name
+	integration.ConstructsRef = params.Refs
+
+	existingResource := dag.GetResourceByVertexId(integration.Id().String())
+	if existingResource != nil {
+		graphResource := existingResource.(*ApiIntegration)
+		graphResource.ConstructsRef = append(graphResource.ConstructsRef, params.Refs...)
+	} else {
+		err := dag.CreateDependencies(integration, map[string]any{
+			"RestApi": RestApiCreateParams{
+				Refs: params.Refs,
+				Name: params.ApiName,
+			},
+			"Resource": ApiResourceCreateParams{
+				AppName: params.AppName,
+				Refs:    params.Refs,
+				Name:    params.Name,
+				ApiName: params.ApiName,
+			},
+			"Method": params,
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type ApiMethodCreateParams struct {
+	AppName    string
+	Refs       []core.AnnotationKey
+	Name       string
+	ApiName    string
+	HttpMethod string
+}
+
+// Create takes in an all necessary parameters to generate the RestApi name and ensure that the RestApi is correlated to the constructs which required its creation.
+func (method *ApiMethod) Create(dag *core.ResourceGraph, params ApiMethodCreateParams) error {
+
+	name := apiResourceSanitizer.Apply(fmt.Sprintf("%s-%s-%s", params.AppName, params.Name, params.HttpMethod))
+	method.Name = name
+	method.ConstructsRef = params.Refs
+	method.HttpMethod = params.HttpMethod
+
+	existingResource := dag.GetResource(method.Id())
+	if existingResource != nil {
+		graphResource := existingResource.(*ApiMethod)
+		graphResource.ConstructsRef = append(graphResource.ConstructsRef, params.Refs...)
+	} else {
+		err := dag.CreateDependencies(method, map[string]any{
+			"RestApi": RestApiCreateParams{
+				Refs: params.Refs,
+				Name: params.ApiName,
+			},
+			"Resource": ApiResourceCreateParams{
+				AppName: params.AppName,
+				Refs:    params.Refs,
+				Name:    params.Name,
+				ApiName: params.ApiName,
+			},
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type ApiMethodConfigureParams struct {
+	RequestParameters map[string]bool
+	Authorization     string
+}
+
+// Configure sets the intristic characteristics of a vpc based on parameters passed in
+func (method *ApiMethod) Configure(params ApiMethodConfigureParams) error {
+	method.Authorization = "None"
+	if method.RequestParameters == nil {
+		method.RequestParameters = make(map[string]bool)
+	}
+	if params.RequestParameters != nil {
+		method.RequestParameters = params.RequestParameters
+	}
+	if params.Authorization != "" {
+		method.Authorization = params.Authorization
+	}
```

Comment:
nit: these could just assign to the params value, since it's just checking against the zero value anyway.

Reply:
well those are already set by defaults i think when we do expansion so thats why i do the check. I should likely move that into here though so its more clear, but the request parameters i should then append to the existing list because those we have context of from the expansion func

Reply:
The `params` don't have the defaults in them? I guess I'm confused then, I thought they would.

Reply:
well these wouldnt, i think realistically these shouldnt even be configurable yet because we dont allow it, so i could remove it. Was a little forward looking